core/src/main/scala/org/apache/spark/deploy/master/ui/MasterPage.scala

```diff
@@ -173,6 +177,30 @@ private[spark] class MasterPage(parent: MasterWebUI) extends WebUIPage("") {
       <td>
         {app.coresGranted}
       </td>
+      <td>
+        {app.requestedCores}
+      </td>
+      <td sorttable_customkey={app.desc.memoryPerSlave.toString}>
+        {Utils.megabytesToString(app.desc.memoryPerSlave)}
+      </td>
+      <td>{UIUtils.formatDate(app.submitDate)}</td>
+      <td>{app.desc.user}</td>
+      <td>{app.state.toString}</td>
+      <td>{UIUtils.formatDuration(app.duration)}</td>
+    </tr>
+  }
+
+  private def completeAppRow(app: ApplicationInfo): Seq[Node] = {
+    <tr>
```

Comment:
Is there a way to refactor this to not duplicate most of the row markup? like save off two other `Seq[Node]` of the common prefix and suffix?